rich/live.py

```diff
@@ -0,0 +1,347 @@
+                        self.ipy_widget = Output()
+                        display(self.ipy_widget)
+
+                    with self.ipy_widget:
+                        self.ipy_widget.clear_output(wait=True)
+                        self.console.print(self._live_render.renderable)
+        elif self.console.is_terminal and not self.console.is_dumb_terminal:
+            with self._lock, self.console:
+                self.console.print(Control(""))
+        elif (
+            not self.is_started and not self.transient
+        ):  # if it is finished allow files or dumb-terminals to see final result
+            with self.console:
+                self.console.print(Control(""))
+
+    def _disable_redirect_io(self):
+        """Disable redirecting of stdout / stderr."""
+        if self._restore_stdout:
+            sys.stdout = self._restore_stdout
+            self._restore_stdout = None
+        if self._restore_stderr:
+            sys.stderr = self._restore_stderr
+            self._restore_stderr = None
+
+    def process_renderables(
+        self, renderables: List[ConsoleRenderable]
+    ) -> List[ConsoleRenderable]:
+        """Process renderables to restore cursor and display progress."""
+        if self.console.is_terminal:
+            # lock needs acquiring as user can modify live_render renerable at any time unlike in Progress.
+            with self._lock:
+                # determine the control command needed to clear previous rendering
+                clear_control = self._clear_render.position_cursor()
+
+                # on non-transient allow the final live render to be displayed
+                # check that renderable doesn't overflow terminal height or it will
+                if self.hide_overflow and self._started:
+                    # determine height of renderable
+                    lines = self.console.render_lines(
+                        self._live_render.renderable, pad=False
+                    )
+                    renderable_height = Segment.get_shape(lines)[1]
+                    if renderable_height >= self.console.size.height:
+                        # continued overflow re-render terminal too small
+                        self._clear_render = self._hide_render
+                        return [
+                            clear_control,
+                            *renderables,
+                            self._hide_render,
+                        ]
+                # new clear
+                self._clear_render = self._live_render
+                return [
+                    clear_control,
+                    *renderables,
+                    self._live_render,
+                ]
+        elif (
+            not self.is_started and not self.transient
+        ):  # if it is finished render the final output for files or dumb_terminals
+            return [*renderables, self._live_render]
+
+        return renderables
+
+
+if __name__ == "__main__":
```

Comment:
You might want to add a `# pragma: no cover` here to exclude the block from coverage.